Scope of disable_active_migration (#3670)
quicwg/base-drafts: draft-ietf-quic-transport.md

  transport parameter to request that clients move connections to that dedicated
  address. Note that clients could choose not to use the preferred address.

A server in a deployment that does not implement a solution to maintain
connection continuity when the client address changes SHOULD indicate migration
is not supported using the disable_active_migration transport parameter.  The
disable_active_migration transport parameter does not prohibit connection

> migration after a client has acted on the preferred_address transport parameter.

Reviewer: ```suggestion
migration after a client has acted on a preferred_address transport parameter.
```

  transport parameter to request that clients move connections to that dedicated
  address. Note that clients could choose not to use the preferred address.

A server in a deployment that does not implement a solution to maintain

> connection continuity during connection migration SHOULD disallow migration

Reviewer 2: ```suggestion
connection continuity when the client address changes SHOULD indicate migration is not supported
```

using the disable_active_migration transport parameter.  The

> disable_active_migration transport parameter does not prohibit subsequent

Reviewer 2: ```suggestion
disable_active_migration transport parameter does not prohibit
```

duration of the handshake.  An endpoint MUST NOT initiate connection migration
before the handshake is confirmed, as defined in section 4.1.2 of {{QUIC-TLS}}.

If the peer sent the disable_active_migration transport parameter, an endpoint
also MUST NOT send packets (including probing packets; see {{probing}}) from a
different local address to the address the peer used during the handshake. An

> endpoint which has sent this transport parameter, but detects that a peer has

Reviewer: ```suggestion
endpoint that has sent this transport parameter, but detects that a peer has
```

> nonetheless migrated to a different network MUST either drop the incoming

Reviewer: ```suggestion
nonetheless migrated to a different remote address MUST either drop the incoming
```

disable_active_migration (0x0c):

: The disable active migration transport parameter is included if the endpoint
  does not support active connection migration ({{migration}}) on the address
  being used during the handshake.  When a peer sets this transport parameter,

>   the recipient MUST NOT send any packets to the address the peer used to
>   perform the handshake from a local address or port other than that used to
>   perform the handshake.  This parameter is a zero-length value.

Reviewer: I *might have a suggestion.
```suggestion
  an endpoint MUST NOT use a new local address when sending to the address
  that the peer used during the handshake.  This parameter is a zero-length value.
```

>   perform the handshake.  This parameter is a zero-length value.

Reviewer 3: I would reiterate the sentence from above here:
> The disable_active_migration transport parameter does not prohibit connection migration after a client has acted on the preferred_address transport parameter.

I fear that otherwise folks might forget that detail as it is very much not clear from the name of the parameter

disable_active_migration (0x0c):

: The disable active migration transport parameter is included if the endpoint
  does not support active connection migration ({{migration}}) on the address
  being used during the handshake.  Peers of an endpoint that sets this
  transport parameter MUST NOT send any packets, including probing packets
  ({{probing}}), from a local address or port other than that used to perform

>   the handshake. This parameter is a zero-length value.

Reviewer 4: To me it seems that we need to also change this sentence (starting from "Peers ..."). We need to tighten the restriction to only when the client sends the packets to the original address of the server.

Maybe something like: _Peers of an endpoint that sets this transport parameter MUST NOT send any packets, including probing packets ({{probing}}) __to the address that was used to perform the handshake__, from a local address or port other than that used to perform the handshake._

Reviewer: Good point (I made this same point verbally, but failed to notice the absence of a fix).

Author: I toyed with that and the existing sentence already has so many sub-phrases that it became too unwieldy.  I'll see if I can split it up any better.
